ambry-frontend/src/test/java/com.github.ambry.frontend/AmbrySecurityServiceTest.java

```diff
@@ -299,20 +302,54 @@ private void verifyAbsenceOfHeaders(MockRestResponseChannel restResponseChannel,
 
   /**
    * Tests {@link SecurityService#processResponse(RestRequest, RestResponseChannel, BlobInfo, Callback)} for a Get blob
-   * with the passed in {@link BlobInfo}
+   * with the passed in {@link BlobInfo} and various range settings, including no set range (entire blob).
+   * @param blobInfo the {@link BlobInfo} to be used for the {@link RestRequest}s
+   * @throws Exception
+   */
+  private void testGetBlobWithVariousRanges(BlobInfo blobInfo)
+      throws Exception {
+    long blobSize = blobInfo.getBlobProperties().getBlobSize();
+    testGetBlob(blobInfo, null);
+    testGetBlob(blobInfo, ByteRange.fromStartOffset(ThreadLocalRandom.current().nextLong(blobSize)));
```

Comment:
And in all the other cases - would be good to make sure they don't match the base case.